lk2nd/lk2nd-device.c

```diff
@@ -275,6 +275,34 @@ static void lk2nd_parse_device_node(const void *fdt)
 #endif
 }
 
+struct lk2nd_keymap* lk2nd_fdt_parse_early_keys(void)
```

Comment:
Can we make this not early(TM) somehow? Would prefer not calling `lk2nd_find_device_offset` more often than necessary. I thought that's why you moved `lk2nd_init()`. You should parse the `struct lk2nd_keymap` into `struct lk2nd_device` and then use it from there.